### PR TITLE
Torchrec Enablement Fixes - Re-PR 107910

### PR DESCRIPTION
**Pull request by voznesenskym**
cc @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

### CODE REVIEW

File: torch/_dynamo/variables/misc.py [hunk 1]
```diff
@@ -714,6 +714,43 @@ def call_function(
...
+            for item in seq:
+                if acc is None:
+                    acc = item
+                else:
+                    try:
+                        acc = func(tx, item, acc)
+                    except Exception:
+                        raise unimplemented(
+                            f"Unexpected failure in invoking function during accumulate. Failed running func {func}({item}{acc})"
+                        )
```
Comment (Skylion007) [inline torch/_dynamo/variables/misc.py:748]:
```suggestion
                    except Exception as ex:
                        raise unimplemented(
                            f"Unexpected failure in invoking function during accumulate. Failed running func {func}({item}{acc})"
                        ) from ex

Comment (Skylion007) [reply torch/_dynamo/variables/misc.py:748]: Catching a bare exception is really frowned upon in general here though, I would recommend trying to limit what excetion you are catching signficantly. Is it a RuntimeException? and UnimplementedException etc...

**Review by wconstab — APPROVED** [review]
restamp

**Pull request closed without merging**